client/my-sites/plans/jetpack-plans/upsell/index.tsx

```diff
@@ -0,0 +1,176 @@
+import slugToSelectorProduct from '../slug-to-selector-product';
+import type { QueryArgs } from 'calypso/my-sites/plans/jetpack-plans/types';
+
+import './style.scss';
+
+interface Props {
+	siteSlug: string;
+	productSlug: string;
+	urlQueryArgs: QueryArgs;
+}
+
+const JetpackUpsellPage: React.FC< Props > = ( { siteSlug, productSlug, urlQueryArgs } ) => {
+	const upsellSlug = PURCHASE_FLOW_UPSELLS_MATRIX[ productSlug ];
+	const productNames = getJetpackProductsDisplayNames();
+
+	const [ isLoadingUpsellPageExperiment, experimentAssignment ] = useExperiment(
+		'jetpack_upsell_page_2022_05'
+	);
+
+	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const isFetchingPurchases = useSelector( isFetchingSitePurchases );
+	const purchases = useSelector( ( state ) => getSitePurchases( state, siteId ) );
+
+	const productItem = useMemo( () => slugToSelectorProduct( productSlug ), [ productSlug ] );
+	const upsellItem = useMemo(
+		() => ( upsellSlug ? slugToSelectorProduct( upsellSlug ) : null ),
+		[ upsellSlug ]
+	);
+	const productCheckoutURL = useMemo(
+		() => buildCheckoutURL( siteSlug, productSlug, urlQueryArgs ),
+		[ siteSlug, productSlug, urlQueryArgs ]
+	);
+	const upsellCheckoutURL = useMemo(
+		() => ( upsellSlug ? buildCheckoutURL( siteSlug, upsellSlug, urlQueryArgs ) : '' ),
+		[ siteSlug, upsellSlug, urlQueryArgs ]
+	);
+	const isUpsellOwned = useMemo(
+		() =>
+			isFetchingPurchases ? false : !! purchases?.find( ( p ) => p.productSlug === upsellSlug ),
+		[ upsellSlug, purchases, isFetchingPurchases ]
+	);
+	const { intro, description, cta, features } = useMemo( () => {
+		if ( PLAN_JETPACK_SECURITY_T1_YEARLY === upsellSlug ) {
+			return {
+				intro: translate( 'Here’s a popular bundle for comprehensive site security:' ),
+				description: translate( 'Save money with all Jetpack security products in one bundle:' ),
+				cta: translate( 'Upgrade your site security for only an additional:' ),
+				features: [
+					{
+						slug: PRODUCT_JETPACK_BACKUP_T1_YEARLY,
+						text: translate( 'Real-time cloud backups and one-click restores' ),
+					},
+					{
+						slug: PRODUCT_JETPACK_SCAN,
+						text: translate( 'Malware scanning and one-click fixes' ),
+					},
+					{
+						slug: PRODUCT_JETPACK_ANTI_SPAM,
+						text: translate( 'Block spam in comments and forms' ),
+					},
+				],
+			};
+		}
+
+		return {};
+	}, [ upsellSlug ] );
+
+	const productName = productItem?.displayName;
+	const upsellName = upsellItem?.displayName;
+
+	const productPriceObj = useItemPrice( siteId, productItem );
+	const upsellPriceObj = useItemPrice( siteId, upsellItem );
+	const priceDelta =
+		( upsellPriceObj?.discountedPrice || upsellPriceObj?.originalPrice ) -
+		( productPriceObj?.discountedPrice || productPriceObj?.originalPrice );
+	const isLoadingPrice = productPriceObj?.isFetching || upsellPriceObj?.isFetching;
+
+	useEffect( () => {
```

Comment:
Redirect to checkout page if:
- User is not assigned the treatment variation
- The product to upsell is not a valid product to upsell
- Site has already a subscription to the upsold product